packages/smooth_app/lib/pages/product/common/product_list_widget.dart

```diff
@@ -0,0 +1,147 @@
+      const int DAY_IN_MILLIS = 24 * 3600 * 1000;
+      String? daysAgoLabel;
+      for (final Product product in products) {
+        final int timestamp = productExtras[product.barcode]!.intValue;
+        final int daysAgo = ((nowInMillis - timestamp) / DAY_IN_MILLIS).round();
+        final String tmpDaysAgoLabel = _getDaysAgoLabel(daysAgo);
+        if (daysAgoLabel != tmpDaysAgoLabel) {
+          daysAgoLabel = tmpDaysAgoLabel;
+          metas.add(_Meta.daysAgoLabel(daysAgoLabel));
+        }
+        metas.add(_Meta.product(product));
+      }
+    } else {
+      for (final Product product in products) {
+        metas.add(_Meta.product(product));
+      }
+    }
+    return ReorderableListView.builder(
+      onReorder: (final int oldIndex, final int newIndex) async {
+        widget.productList.reorder(oldIndex, newIndex);
+        daoProductList
+            .put(widget.productList); // careful: if "await", flickering
+        setState(() {});
+      },
+      buildDefaultDragHandles: false,
+      itemCount: metas.length,
+      itemBuilder: (BuildContext context, int index) {
+        final _Meta meta = metas[index];
+        if (!meta.isProduct()) {
+          return ListTile(
+            key: Key(meta.daysAgoLabel!),
+            leading: const Icon(Icons.history),
+            title: Text(meta.daysAgoLabel!),
+          );
+        }
+        final Product product = meta.product!;
+        final Widget child = Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 12.0, vertical: 8.0),
+          child: ProductListItem(
+            product: product,
+            productList: widget.productList,
+            listRefresher: () => setState(() {}),
+            daoProductList: daoProductList,
+            reorderIndex: widget.reorderable ? index : null,
+          ),
+        );
+        if (widget.dismissible) {
+          return Dismissible(
+            background: Container(color: colorScheme.background),
+            key: Key(product.barcode!),
+            onDismissed: (final DismissDirection direction) async {
+              final bool removed = widget.productList.remove(product.barcode!);
+              if (removed) {
+                await daoProductList.put(widget.productList);
+                setState(() => metas.removeAt(index));
+              }
+              ScaffoldMessenger.of(context).showSnackBar(
+                SnackBar(
+                  content: Text(
+                      removed ? 'Product removed' : 'Could not remove product'),
+                  duration: const Duration(seconds: 3),
+                ),
+              );
+              // TODO(monsieurtanuki): add a snackbar ("put back the food")
+            },
+            child: child,
+          );
+        }
+        return Container(
+          key: Key(product.barcode!),
+          child: child,
+        );
+      },
+    );
+  }
+
+  static String _getDaysAgoLabel(final int daysAgo) {
```

Comment:
While you're refactoring, you'd be better off creating a new file dedicated to "some time ago" labels. Maybe `some_time_ago_helper.dart`, with `String getDaysAgoLabel(final int daysAgo, final AppLocalizations appLocalizations)`. With actual translations, at least in `TODO`s.

Reply:
Done.